src/ErrorHandling/ErrorModal.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { useEffect, useState } from "react";
+import { makeStyles } from "tss-react/mui";
+import { Modal } from "../ui/React/Modal";
+import { Box, Button, Typography } from "@mui/material/";
+import { ErrorRecord, ErrorState } from "./ErrorState";
+import { Router } from "../ui/GameRoot";
+import { SimplePage } from "@enums";
+import { useRerender } from "../ui/React/hooks";
+import { OptionSwitch } from "../ui/React/OptionSwitch";
+import { LogBoxEvents } from "../ui/React/LogBoxManager";
+import { recentScripts } from "../Netscript/RecentScripts";
+
+export function ErrorModal(): React.ReactElement {
+  const { classes } = useStyles();
+  const rerender = useRerender();
+  const [error, setError] = useState<ErrorRecord | null>(ErrorState.ActiveError);
+
+  useEffect(() => {
+    const listener = (newError: ErrorRecord) => {
+      if (newError.force || (Router.page() !== SimplePage.ActiveScripts && !ErrorState.PreventModals)) {
+        setError(newError);
+        rerender();
+      }
+    };
+    ErrorState.ErrorUpdate.subscribe(listener);
+  }, [rerender]);
+
+  const onClose = (): void => {
+    ErrorState.ActiveError = null;
+    setError(null);
+  };
+
+  const viewLogs = (): void => {
+    onClose();
+    const recentScript = recentScripts.find((script) => script.runningScript.pid === error?.pid);
+    if (!recentScript) {
+      console.warn(`No recent script found with pid ${error?.pid}`);
+      return;
+    }
+    LogBoxEvents.emit(recentScript.runningScript);
+  };
```

Comment:
```suggestion
  const viewLogs = (): void => {
    if (error === null) {
      return;
    }
    const recentScript = recentScripts.find((script) => script.runningScript.pid === error.pid);
    if (!recentScript) {
      SnackbarEvents.emit(`No recent script found with pid ${error.pid}`, ToastVariant.INFO, 2000);
      return;
    }
    onClose();
    LogBoxEvents.emit(recentScript.runningScript);
  };
```
- Emit a notification instead of `console.warn()`: The player will be confused when they press a button, but nothing happens.
- Only close this modal if we can open the log modal.